### DISCUSSION TITLE
Pin an individual comment from an issue

### DISCUSSION THREAD

**Discussion by hwittenborn** [community/community#47912] [Projects and Issues] [answered]
### Select Topic Area

General

### Body

Currently you can pin GitHub issues, but not individual comments within those issues. It'd be helpful if I could pin an individual comment so that a helpful comment, such as a fix to an issue that's still being fixed, can be easily found within a comment.

I'm thinking the pinned comment can just be seen when viewing the issue directly - pinned issues can function as they currently do, and then the pinned comment can just be seen once the issue is clicked.

**Comment by cwway** [comment]
Chiming in on this - when using [issues](https://github.com/issues) over time like a product dev/feature request catalogue, comments are added and stack up significantly. 

**The ability to PIN a particular comment, or perhaps overwrite ordering by assigning 1-2-3  priority, would be fantastic to bring at least one over-arching or conclusive comment to the top**

**Reply by cwway** [reply]
### Follow up:

Consider ability to "highlight" any comments (checkbox on the comment or similar UI solution), then at the top of the issue, a simple toggle for 'top comments' tooltip: "show highlighted comments" ... Mock-up:

<img width="959" height="255" alt="image" src="https://github.com/user-attachments/assets/02606ff8-3fd9-4cf8-bc9f-b4f77b07c573" />

Update: just realized this is well described below by @fantasai [here](https://github.com/orgs/community/discussions/47912#discussioncomment-12394933)

**Comment by zadjii-msft** [comment]
Just to x-ref:
* #47912
* #27522
* #18523

all seem like they're the same request.

**Reply by Marcono1234** [reply]
And also from the now archived `isaacs/github` repository: https://github.com/isaacs/github/issues/1008
Which contains some examples and references to related feature requests.

**Reply by gnowland** [reply]
And https://github.com/isaacs/github/issues/1555 from the now archived `isaacs/github` repository.

**Reply by Marcono1234** [reply]
That issue from the `isaacs/github` repository seems to be about comments on PRs. For that there are other also other related suggestions here:
- #11842
- #7638

Though maybe having the option to pin comments on PRs could be useful nonetheless, but I am not sure if it would solve the issues mentioned in those discussions.

**Comment by gnowland** [comment]
The biggest issue here is when an important "current progress/blockers" comment gets hidden under "Load more..." leading to people constantly asking the current state. Pinning a comment should highlight it in some way as well as prevent it from being collapsed. 

Heck, it might even make sense to collapse everything \*up to\* the pinned comment, since that comment is often the new jumping off point for  further discussion on long-open issues.

**Comment by oleg-rswll** [comment]
This would be a really useful feature to have.  Is there any status on when it may be available?

**Comment by Joao-Pedro-P-Holanda** [comment]
This feature of pinning comments can also be useful inside discussions, that naturally have more interactions than most issues. In a long term discussion (e.g. a discussion to keep track of the translation of a project's documentation) is important to have a way to keep the most relevant decisions or guidelines at the top.

**Comment by fantasai** [comment]
Came here to ask even if we could just mark or highlight certain comments as key comments. Sometimes there's multiple of these. But they get lost in long discussions, and can even be hidden under "load more".

Here's an example: https://github.com/w3c/csswg-drafts/issues/6245#issuecomment-2661545284 This comment is actually a good candidate for @gnowland's ["collapse up until pin" idea](https://github.com/orgs/community/discussions/47912#discussioncomment-6816029), since it resets the conversation. But in many issues there are multiple key points that anyone working on the issue should review, and it's good to not lose those, either. In this issue we actually thought we had solved it until we went back and reviewed key comments in the discussion, which prompted redesigning the whole thing.

**Reply by alinnert** [reply]
[This issue](https://github.com/nuxt/nuxt/issues/31978) is another example. All of the sudden many JS framework dev servers print a warning in the terminal that something is making a request to the URL `/.well-known/appspecific/com.chrome.devtools.json` but it results in a 404. Everyone was confused, people thought some installed extension is responsible for this request.

Right in the middle of this conversation there is [the answer to why this happens](https://github.com/nuxt/nuxt/issues/31978#issuecomment-2869622575): The Chrome DevTools team added the feature "Automatic Workspace Folders" which is sending this request for any dev server on `localhost` to provide information about the current project. This changes the context of this issue completely. But, of course, it took me a while to find it.

And this is not the first time I've run into this kind of situation.

**Comment by NarendraJoglekar** [comment]
This would be a really useful feature to have. Is there any status on when it may be available?

**Comment by azenMatt** [comment, marked as answer]
This has now shipped! https://github.com/orgs/community/discussions/186435
